breadbox/breadbox/schemas/context.py

```diff
@@ -19,4 +24,15 @@ class Context(BaseModel):
     # - { "==": [ {"var": "slice/growth_pattern/all/label"}, "Adherent" ] }
     # - True
     expr: ContextExpression
-    context_type: str  # Dimension type, Ex. "depmap_model"
+    name: Optional[str] = None
+
+    # The "dimension_type" field is used by contexts in the new format (with slice queries)
+    # The "context_type" field is used by the old format (with slice IDs)
+    # The two fields contain the same information (the dimension type name)
+    dimension_type: Optional[str] = None
+    context_type: Optional[str] = None
+    vars: dict[str, dict[str, str]] = {}
+
+
+class ContextRequest(BaseModel):
```

Comment:
Is there a reason why we need to create a new type `ContextRequest`? It seems like ContextRequest is used to type the request parameter but we are ultimately only interested in accessing contents in its one and only field `context`. Why not just type the parameter as `Context` or rename it to `ContextRequest`?

Reply:
Alternatively, you could also do something like:
```
class ContextRequest(Context):
    pass
```

Reply:
Good question. I agree that this is not a totally normal way to model a request schema. I only added this outer wrapper because I wanted to be able to support the exact same legacy request format we've been using for the past year or so (in addition to the newer, more-flexible, easier-to-read format). 

The reason I'm hesitant to change the format is because these "contexts" are stored in cookies in the users' browsers. I'm not actually 100% sure whether this full object is stored as a cookie, or only the inner dictionary, but regardless I'm hesitant to make Randy do even more refactoring unless there's a clear reason for it. My goal with these new context endpoints was to:
1. Continue to support the old "context" request format
2. Also support a new request format, which is more flexible and supports tabular lookups
3. Return a response which includes more information (both IDs and labels)

For example, a legacy context request is formatted like:
```
{"context": {
    "context_type": "depmap_model",
    "expr": {">": [{"var": f"slice/{dataset_id}/gene1/label"}, 0.05]},
}}
```

Instead of (what I'm guessing you're expecting):
```
{
    "context_type": "depmap_model",
    "expr": {">": [{"var": f"slice/{dataset_id}/gene1/label"}, 0.05]},
}
```
I agree that the cleanest thing to do would be to update it, I'm just not sure if it's worth the effort of refactoring the things that use the old format.